spec/requests/application_controller/filter_spec.rb

```diff
@@ -0,0 +1,120 @@
+      },
+      :status      => 400,
+    )
+  end
+
+  def expect_success(query, *results)
+    get("/api/v0.1/tasks?#{query}")
+
+    expect(response).to have_attributes(
+      :parsed_body => paginated_response(results.length, results.collect { |i| a_hash_including("id" => i.id.to_s) }),
+      :status      => 200,
+    )
+  end
+
+  context "strings" do
+    let!(:task_1) { create_task(:name => "aaa") }
+    let!(:task_2) { create_task(:name => "bbb") }
+    let!(:task_3) { create_task(:name => "abc") }
+    let!(:task_4) { create_task }
+
+    it("key:eq single without 'eq' key")          { expect_success("filter[name]=#{task_1.name}", task_1) }
+    it("key:eq array of values without 'eq' key") { expect_success("filter[name][]=#{task_1.name}&filter[name][]=#{task_2.name}", task_1, task_2) }
+    it("key:eq single with 'eq' key")             { expect_success("filter[name][eq]=#{task_1.name}", task_1) }
+    it("key:eq array of values with 'eq' key")    { expect_success("filter[name][eq][]=#{task_1.name}&filter[name][eq][]=#{task_2.name}", task_1, task_2) }
+
+    it("key:contains single")                     { expect_success("filter[name][contains]=a", task_1, task_3) }
+    it("key:contains array")                      { expect_success("filter[name][contains][]=a&filter[name][contains][]=b", task_3) }
+
+    it("key:ends_with")                           { expect_success("filter[name][ends_with]=a", task_1) }
+    it("key:starts_with")                         { expect_success("filter[name][starts_with]=b", task_2) }
+    it("key:nil")                                 { expect_success("filter[name][nil]", task_4) }
+    it("key:not_nil")                             { expect_success("filter[name][not_nil]", task_1, task_2, task_3) }
+  end
+
+  context "integers" do
+    let!(:task_1) { create_task }
+    let!(:task_2) { create_task }
+    let!(:task_3) { create_task }
+    let!(:task_4) { create_task }
+
+    it("key:eq single without 'eq' key")          { expect_success("filter[id]=#{task_1.id}", task_1) }
+    it("key:eq array of values without 'eq' key") { expect_success("filter[id][]=#{task_1.id}&filter[id][]=#{task_2.id}", task_1, task_2) }
+    it("single with 'eq' key")                    { expect_success("filter[id][eq]=#{task_1.id}", task_1) }
+    it("array of values with 'eq' key")           { expect_success("filter[id][eq][]=#{task_1.id}&filter[id][eq][]=#{task_2.id}", task_1, task_2) }
+
+    it("key:gt")                                  { expect_success("filter[id][gt]=#{task_2.id}", task_3, task_4) }
+    it("key:gte")                                 { expect_success("filter[id][gte]=#{task_2.id}", task_2, task_3, task_4) }
+    it("key:lt")                                  { expect_success("filter[id][lt]=#{task_3.id}", task_1, task_2) }
+    it("key:lte")                                 { expect_success("filter[id][lte]=#{task_3.id}", task_1, task_2, task_3) }
+  end
+
+  context "timestamps" do
```

Comment:
do we support gt, gte, etc for timestamps?

Reply:
Yes , all of the Integer comparisons listed here.  https://github.com/ManageIQ/topological_inventory-api/pull/156/files#diff-08a3013ef490c3273c6b0ffc7c84a0eaR3